fontbe/src/glyphs.rs

```diff
@@ -22,6 +29,86 @@ pub fn create_glyph_work(glyph_name: GlyphName) -> Box<BeWork> {
     Box::new(GlyphWork { glyph_name })
 }
 
+fn create_component(
+    context: &Context,
+    ref_glyph_name: &GlyphName,
+    transform: &Affine,
+) -> Result<(Component, Bbox), GlyphProblem> {
+    // Obtain glyph id from static metadata
+    let gid = context
+        .ir
+        .get_static_metadata()
+        .glyph_id(ref_glyph_name)
+        .ok_or(GlyphProblem::NotInGlyphOrder)?;
+    let gid = GlyphId::new(gid as u16);
+
+    // No known source does point anchoring so we just our transform into a 2x2 + offset
+    let [a, b, c, d, e, f] = transform.as_coeffs();
+    let component = Component::new(
+        gid,
+        Anchor::Offset {
+            x: e as i16,
+            y: f as i16,
+        },
+        Transform {
+            xx: F2Dot14::from_f32(a as f32),
+            yx: F2Dot14::from_f32(b as f32),
+            xy: F2Dot14::from_f32(c as f32),
+            yy: F2Dot14::from_f32(d as f32),
+        },
+        ComponentFlags::default(),
+    );
+
+    // Bbox computation is postponed to glyph merge to ensure all glyphs are available to query
+    Ok((component, Bbox::default()))
+}
+
+fn create_composite(
+    context: &Context,
+    glyph_name: &GlyphName,
+    default_location: &NormalizedLocation,
+    components: &HashMap<(GlyphName, NormalizedLocation), Affine>,
```

Comment:
why is this `components` an HashMap and not just a Vec of tuples? It is entirely possible that for a given location, more than one component with the same `ref_glyph_name` exist (they'd likely have a different affine, otherwise they'd exactly overlap one another, which is absurd though still valid)

Reply:
Oh that's an interesting point. Maybe I should have a set of (name, loc, affine).

Reply:
I will add a test case that shows off the bug here

Reply:
Got a lovely test case where a component goes missing now :)

Reply:
A set would dedup but would lose the order. I think we also want to keep the arbitrary user-defined ordering components within a given composite glyph, hence why I suggested Vec. The order of the components defines the the control points indices in a composite glyph, that can be referenced from truetype hinting instructions (we don't support them now, but you never know).
Also I mean, if one wants to shoot themselves in the feet by defining the exact same component twice (never ever seen that in the wild by the way) why should we not let them, garbage in, garbage out.

Reply:
I think you should not prune duplicate components, that's not currently happening in the python pipeline and may be unexpected; also it is conceivable to devise a use case for that; e.g. two components that are on top of one another in some but not all masters; or they are shifted only by truetype instructions. In general, let's try to keep faithful to the sources whenever we can.

Reply:
In coding it up I concluded the same, plus I believe it might in future - boring expansion - be valid to have overlapping components at one normalized location but not another.

Reply:
if by irrelevant you mean they would display the same because fill rules don't care about the relative ordering of various subpaths, then yes -- at least, provided the composite glyph is unhinted or contain no TT glyph instruction directly referencing point indices, in which case shuffling components around would break those TT instructions

Reply:
Fixed, is now a vec

Reply:
plus, both our source formats define glyph components as lists with a user-defined ordering, it makes more sense to just record that order in a Vec and keep that in glyf composites, as expected by user.